bot/exts/moderation/slowmode.py

```diff
@@ -42,17 +53,29 @@ async def get_slowmode(self, ctx: Context, channel: MessageHolder) -> None:
             channel = ctx.channel
 
         humanized_delay = time.humanize_delta(seconds=channel.slowmode_delay)
-
-        await ctx.send(f"The slowmode delay for {channel.mention} is {humanized_delay}.")
+        if await self.slowmode_expiration_cache.contains(channel.id):
+            expiration_time = await self.slowmode_expiration_cache.get(channel.id)
+            expiration_timestamp = discord_timestamp(expiration_time, TimestampFormats.RELATIVE)
+            await ctx.send(
+                f"The slowmode delay for {channel.mention} is {humanized_delay} and expires in {expiration_timestamp}."
+            )
+        else:
+            await ctx.send(f"The slowmode delay for {channel.mention} is {humanized_delay}.")
 
     @slowmode_group.command(name="set", aliases=["s"])
     async def set_slowmode(
         self,
         ctx: Context,
         channel: MessageHolder,
         delay: DurationDelta | Literal["0s", "0seconds"],
+        duration: DurationDelta | None = None
```

Comment:
Use `Duration` instead of `DurationDelta`, which handles the conversion to datetime for you.